mdb.py

```
@@ -226,14 +688,742 @@ def execute_change(db, server, sql):
+                    col_index = content['column_names'].index(pk_col)
+                    pk_value = row[col_index]
+
+                    # Handle NULL values
+                    if pk_value is None:
+                        where_conditions.append(f"{pk_col} IS NULL")
+                    else:
+                        # Escape single quotes in values
+                        escaped_value = str(pk_value).replace("'", "''")
+                        where_conditions.append(f"{pk_col} = '{escaped_value}'")
+                except (ValueError, IndexError) as e:
+                    logging.error(f"Error processing primary key column {pk_col}: {e}")
```

Ruff (0.15.4)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)